bootstrap/pkg/apis/apps/kfconfig/types.go

```diff
@@ -514,23 +516,43 @@ func (c *KfConfig) SyncCache() error {
 	return nil
 }
 
-// GetSecret returns the specified secret or an error if the secret isn't specified.
-func (c *KfConfig) GetSecret(name string) (string, error) {
+// Base64Encrypt encrypts a string to based64 encoding.
+func Base64Encrypt(password string) (string, error) {
```

Comment:
Why do this here instead of gcp.go?  Bcrypt encoding secrets is specific to the basic auth secret.
Now that we got rid of HashedLiteralSource in v1beta1  its no longer a part of the generic API for secrets
https://github.com/kubeflow/kubeflow/blob/f943808a2805d81c90935f77b89b8bb85e43baa0/bootstrap/pkg/apis/apps/kfdef/v1beta1/application_types.go#L89

If we make it a public method people will likely use it leading to problems.